addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -76,6 +76,42 @@ var TranslatableFieldMixin = {
 
 var DynamicPlaceholderFieldMixin = {
     DYNAMIC_PLACEHOLDER_TRIGGER_KEY: '#',
+    /**
+     * Overridable method that ensure the modelSelectorPopover is
+     * inserted at the right place in the DOM and positioned properly.
+     *
+     * @public
+     * @param {ModelFieldSelectorPopover} modelSelector     *
+     * @returns {Promise<ModelFieldSelectorPopover>}
+     */
+    insertAndPositionModelSelector: async function (modelSelector) {
+        const $insertAfterElement = this.$el.first();
+        const $modalParent = $insertAfterElement.parents('div.modal-content');
```

Comment:
In particular this should be `closest` rather than parent, unless `$insertAfterElement` can be a text node ?